### ISSUE TITLE
"there is nothing to install in `uniffi_bindgen v0.25.1`"

### ISSUE THREAD

**Issue by frebib** [mozilla/uniffi-rs#1841]
I'm a user that's mostly unfamiliar with Rust trying to build matrix-rust-sdk https://github.com/matrix-org/matrix-rust-sdk/tree/main/bindings/matrix-sdk-ffi

I started reading their sparse documentation, and after a bit of intuition worked out that I needed `uniffi-bindgen` to generate the python bindings. I managed to scramble through the pretty user-unfriendly uniffi documentation and find this telling me to run `cargo install uniffi_bindgen` https://mozilla.github.io/uniffi-rs/internals/api/uniffi_bindgen/index.html#3-generate-and-include-component-scaffolding-from-the-udl-file

I've not been having much luck. In a fresh `rust:latest` docker container, it simply doesn't work. I'm not Rust-literate enough to be able to read between the lines of this warning.

```
Status: Downloaded newer image for rust:latest
docker.io/library/rust:latest
root@7a327d6b390b:/# cargo install uniffi_bindgen
    Updating crates.io index
  Downloaded uniffi_bindgen v0.25.1
  Downloaded 1 crate (130.7 KB) in 0.21s
error: there is nothing to install in `uniffi_bindgen v0.25.1`, because it has no binaries
`cargo install` is only for installing programs, and can't be used with libraries.
To use a library crate, add it as a dependency to a Cargo project with `cargo add`.
```

To summarise, there are two major issues here
- The documentation is not clear enough if you're a user that just happens to need to use uniffi-bindgen to generate bindings. I reckon a "click here if all you want to do is generate bindings" in the README would probably suffice
- I still have no idea how to install the `uniffi-bindgen` binary despite following the documentation closely.

**Comment by mhammond** [comment]
Sorry, those docs are out of date. You need to create your bindings as described at https://mozilla.github.io/uniffi-rs/tutorial/foreign_language_bindings.html (or in my PR)

There's a fair chance the tool you want to use comes with a binary, which you can probably run via `cargo run --bin -p uniffi-bindgen`.
